ui/lib/sync/addon/components/secrets/page/destinations/create-and-edit.ts

```diff
@@ -71,10 +71,19 @@ export default class DestinationsCreateForm extends Component<Args> {
 
     if (isValid) {
       try {
-        const verb = destination.isNew ? 'created' : 'updated';
-        yield destination.save();
-        this.flashMessages.success(`Successfully ${verb} the destination ${destination.name}`);
-        this.store.clearDataset('sync/destination');
+        // we only want to save if there are changes
+        if (destination.dirtyType as unknown as string) {
+          const verb = destination.isNew ? 'created' : 'updated';
+          yield destination.save();
+          // when saving a record the server returns all credentials as ******
+          // Ember Data observes this as a change, marks the model as dirty and the field will be returned from changedAttributes
+          // if the user then attempts to update the record the credential will get overwritten with the masked placeholder value
+          // since the record will be fetched from the details route we can safely unload it to avoid the aforementioned issue
+          destination.unloadRecord();
+          this.flashMessages.success(`Successfully ${verb} the destination ${destination.name}`);
```

Comment:
Yes it's still accessible here but just in case something changes in a future version where the object may be deleted synchronously I'll go ahead and move the flashMessage above the unload.